libecl/src/ecl_grid.c

```diff
@@ -3777,145 +3894,67 @@ bool ecl_grid_compare(const ecl_grid_type * g1 , const ecl_grid_type * g2 , bool
-      if (i == (ecl_grid->nx - 1))
-        return true;
-      else
-        return false;
-    }
+  for (int plane_nr = 0; plane_nr < 12; plane_nr++) {
+    p0 = &cell->corner_list[ tetrahedron_permutations[ method ][plane_nr][0] ];
+    p1 = &cell->corner_list[ tetrahedron_permutations[ method ][plane_nr][1] ];
+    p2 = &cell->corner_list[ tetrahedron_permutations[ method ][plane_nr][2] ];
 
-    if (point_equal( &p , &cell->corner_list[2])) {
-      if (j == (ecl_grid->ny - 1))
-        return true;
-      else
-        return false;
-    }
-
-    if (point_equal( &p , &cell->corner_list[3])) {
-      if ((j == (ecl_grid->ny - 1)) &&
-          (i == (ecl_grid->nx - 1)))
-        return true;
-      else
-        return false;
-    }
-
-    if (point_equal( &p , &cell->corner_list[4])) {
-      if (k == (ecl_grid->nz - 1))
-        return true;
-      else
-        return false;
-    }
-
-    if (point_equal( &p , &cell->corner_list[5] )) {
-      if ((i == (ecl_grid->nx - 1)) &&
-          (k == (ecl_grid->nz - 1)))
-        return true;
-      else
-        return false;
-    }
-
-    if (point_equal( &p , &cell->corner_list[6] )) {
-      if ((j == (ecl_grid->ny - 1)) &&
-          (k == (ecl_grid->nz - 1)))
-        return true;
-      else
-        return false;
-    }
-
-    if (point_equal( &p , &cell->corner_list[7] )) {
-      if ((i == (ecl_grid->nx - 1)) &&
-          (j == (ecl_grid->ny - 1)) &&
-          (k == (ecl_grid->nz - 1)))
-        return true;
-      else
-        return false;
-    }
-
-    if (ecl_cell_get_twist(cell) > 0) {
-      fprintf(stderr, "** Warning: Point (%g,%g,%g) is in vicinity of twisted cell: (%d,%d,%d) - function:%s might be mistaken.\n", x,y,z,i,j,k, __func__);
-      return false;
-    }
-
-    {
-      double signed_volume = ecl_cell_get_signed_volume( cell );
-      if (fabs( signed_volume) > min_volume) {
-        double sign = 1.0;
-        point_type * p0;
-        point_type * p1;
-        point_type * p2;
-        int phase0 = 0;
-        int method = (phase0 + i + j + k) % 2;
-
-        if (signed_volume < 0)
-          sign = -1;
-        {
-          for (int plane_nr = 0; plane_nr < 12; plane_nr++) {
-
-            p0 = &cell->corner_list[ tetrahedron_permutations[ method ][plane_nr][0] ];
-            p1 = &cell->corner_list[ tetrahedron_permutations[ method ][plane_nr][1] ];
-            p2 = &cell->corner_list[ tetrahedron_permutations[ method ][plane_nr][2] ];
-
-            if (point_equal(p0, p1) || point_equal(p0,p2) || point_equal(p1,p2))
-              continue;
-
-            if (sign * point3_plane_distance(p0 , p1 , p2 , &p ) < 0)
-              return false;
-          }
-          return true;
-        }
-      }
+    if (point_equal(p0, p1) || point_equal(p0,p2) || point_equal(p1,p2))
+      continue;
 
+    if (sign * point3_plane_distance(p0 , p1 , p2 , &p ) < 0)
       return false;
-    }
   }
-}
 
+  return true;
+}
 
+#undef NOT_ON_FACE
```

Comment:
The symbols `NOT_ON_FACE, BELONGS_TO_CELL` and `BELONGS_TO_OTHER` are defined as an enum - with file scope, `#undef` here has no consequences - essentially a bug; alternatively you could use 
```C
#define NOT_ON_FACE  1
#define BELONGS_TO_CELL 2
#define BEONGS_TO_OTHER 3
```
where the enum definition is now.